src/duration.jl

```diff
@@ -1,62 +1,52 @@
 
 """
-    Duration{T}
+    Duration{T} <: Number
 
 A `Duration` represents a period of time, split into an integer number of seconds and a 
 fractional part.
 
 ### Fields
 - `seconds`: The integer number of seconds.
-- `fraction`: The fractional part of the duration, where `T` is a subtype of `Number`.
+- `fraction`: The fractional part of the duration, where `T` is a subtype of `Real`.
 """
-struct Duration{T} 
+struct Duration{T} <: Number 
     seconds::Int 
     fraction::T 
 end
 
-function Duration(seconds::T) where {T<:Number}
-    i, f = divrem(seconds, 1)
-    return Duration{T}(i, f)
+function Duration{T}(seconds::Number) where {T <: Number}
+    i,f = divrem(seconds, 1)
+    return Duration{T}(convert(Int, i), T(f))
 end
 
-function Duration(sec::Int, frac::T) where {T<:Number}
-    return Duration{T}(sec, frac)
-end
+function Duration(seconds::T) where {T <: Number}
+    Duration{T}(seconds)  
+end 
 
+ftype(::Duration{T}) where T = T
```

Comment:
I would change this to:

```
Base.eltype(::Duration{T}) where T = T
```

Reply:
Yes I had actually thought of that, however if you look at the `eltype` description it says: 

> Determine the type of the elements generated by iterating a collection of the given type.

And since `Duration` is neither a collection nor an iterable, it didn't seem right to me to implement it.